master/internal/job/job.go

```diff
@@ -133,6 +141,213 @@ func (j *Jobs) jobQSnapshot(ctx *actor.Context, resourcePool string) (AQueue, er
 	return jobQ, nil
 }
 
+// figure out what changes are needed to move the job to the desired position.
+// Later we can support cross RP moving this way as well.
+func moveJobMessages(
+	jobs []*jobv1.Job,
+	target *jobv1.Job,
+	anchor *jobv1.Job,
+	anchorIdx int,
+	aheadOf bool,
+) (*SetGroupPriority, *MoveJob, error) {
+	// validate anchorIdx
+	if anchorIdx < 0 || anchorIdx > len(jobs) {
+		return nil, nil, fmt.Errorf("invalid anchor index %d", anchorIdx)
+	}
+	// validate anchor and target
+	if anchor == nil {
+		return nil, nil, fmt.Errorf("missing anchor job")
+	}
+	if target == nil {
+		return nil, nil, fmt.Errorf("missing target job")
+	}
+	if target.JobId == anchor.JobId {
+		return nil, nil, nil
+	}
+
+	// sanity check
+	lastJob := int32(-1)
+	for _, job := range jobs {
+		check.Panic(check.GreaterThanOrEqualTo(job.Summary.JobsAhead, lastJob))
+		lastJob = job.Summary.JobsAhead
+	}
+
+	var priorityMsg *SetGroupPriority
+	if target.Priority != anchor.Priority {
+		priority := int(anchor.Priority)
+		priorityMsg = &SetGroupPriority{
+			Priority: &priority,
+		}
+	}
+
+	var moveMsg *MoveJob
+	// find the next or previous job based on aheadOf in the same priority lane
+	var secondAnchor model.JobID
+	if aheadOf {
```

Comment:
Should we do a check for if the job is trying to be inserted in the same position? In which case it would be performing an unnecessary operation.

Reply:
that I believe is being checked after this and I think I wrote a unit test for it as well